app/TuistApp/Sources/Views/DevicesView/DevicesView.swift

```diff
@@ -94,11 +64,98 @@ struct DevicesView: View, ErrorViewHandling {
                     }
                 }
             }
-            .padding([.leading, .trailing], 8)
         }
         .frame(height: isExpanded ? NSScreen.main.map { $0.visibleFrame.size.height - 300 } ?? 500 : nil)
     }
 
+    @ViewBuilder
+    private func pinnedDevicesSection() -> some View {
+        if !viewModel.devices.isEmpty || !viewModel.pinnedSimulators.isEmpty {
+            HStack {
+                Text("Devices")
+                    .font(.headline)
+                    .fontWeight(.medium)
+                    .padding(.leading, 4)
+
+                Spacer()
+
+                Button {
+                    withAnimation(.linear(duration: 0.8)) {
+                        isRefreshing = true
+                        Task {
+                            try await viewModel.onAppear()
+                            await MainActor.run {
+                                isRefreshing = false
+                            }
+                        }
+                    }
+                } label: {
+                    Image(systemName: "arrow.triangle.2.circlepath")
+                        .font(.headline)
+                        .fontWeight(.medium)
+                        .rotationEffect(isRefreshing ? .degrees(360) : .zero)
+                }
+                .buttonStyle(.borderless)
+                .disabled(isRefreshing)
+            }
+
+            Divider()
+
+            let (connectedDevices, disconnectedDevices) = viewModel.devices
+                .reduce(
+                    into: (connected: [PhysicalDevice](), disconnected: [PhysicalDevice]())
+                ) { partialResult, device in
+                    switch device.connectionState {
+                    case .connected:
+                        partialResult.connected.append(device)
+                    case .disconnected:
+                        partialResult.disconnected.append(device)
+                    }
+                }
```

Comment:
I'd move computing the `connectedDevices` and `disconnectedDevices` to the `DevicesViewModel`. That way we can test this logic.